fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/MessageValidatorTest.java

```diff
@@ -189,4 +190,43 @@ public void testValidEmoji() {
     assertThrows(
         IllegalArgumentException.class, () -> validator.isValidEmoji("random string", field));
   }
+
+  @Test
+  public void testValidPopCHAUrl() {
```

Comment:
Entirely outside the scope of this PR, but in general it's better to have separate tests for separate conditions being checked.

E.g., instead of both testing `valid` and `valid2` within this test, you'd have a `testValidPopCHAurlLocalhost` , `testValidPopCHAurlPublicDomain` , and then maybe `testInvalidPopCHAurlMissingClientId`, etc.

This has the effect that when the CI cries about a test having failed, you immediately know which condition failed, without having to track down the stack trace and failing condition.